chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -537,34 +558,63 @@ func (b *BitcoindNotifier) confDetailsManually(confRequest chainntnfs.ConfReques
 		default:
 		}
 
-		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		// Receive the next block hash from the async queue.
+		blockHash, err := batchRequest[height].Receive()
 		if err != nil {
 			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get hash from block "+
-					"with height %d", height)
+				fmt.Errorf("unable to retrieve hash for block "+
+					"with height %d: %v", height, err)
 		}
 
-		block, err := b.GetBlock(blockHash)
-		if err != nil {
+		blockHashAddresses = append(blockHashAddresses, blockHash)
+	}
+
+	// Now we fetch blocks in interval of 15 to avoid out of memory
+	// errors in case of fetching too many blocks with GetBlocksBatch().
+	const batchSize = 15
+	total := len(blockHashAddresses)
+
+	for i := 0; i < total; i += batchSize {
+		// Ensure we haven't been requested to shut down before
+		// processing next set of blocks.
+		select {
+		case <-b.quit:
 			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get block with hash "+
-					"%v: %v", blockHash, err)
+				chainntnfs.ErrChainNotifierShuttingDown
+		default:
 		}
 
-		// For every transaction in the block, check which one matches
-		// our request. If we find one that does, we can dispatch its
-		// confirmation details.
-		for txIndex, tx := range block.Transactions {
-			if !confRequest.MatchesTx(tx) {
-				continue
-			}
+		start := i
+		end := i + batchSize
+
+		if end > total {
+			end = total
+		}
 
-			return &chainntnfs.TxConfirmation{
-				Tx:          tx,
-				BlockHash:   blockHash,
-				BlockHeight: height,
-				TxIndex:     uint32(txIndex),
-			}, chainntnfs.TxFoundManually, nil
+		blocks, err := b.chainConn.GetBlocksBatch(
+			blockHashAddresses[start:end],
+		)
+
+		if err != nil {
+			return nil, chainntnfs.TxNotFoundManually, err
+		}
+
+		height := int(currentHeight) - start
+		for j := range blocks {
+			// For every transaction in the block, check which one
+			// matches our request. If we find one that does, we can
+			// dispatch its confirmation details.
+			for txIndex, tx := range blocks[j].Transactions {
+				if confRequest.MatchesTx(tx) {
+					return &chainntnfs.TxConfirmation{
+						Tx:          tx,
+						BlockHash:   blockHashAddresses[start+j],
+						BlockHeight: uint32(height),
+						TxIndex:     uint32(txIndex),
```

Comment:
nit:  If `BlockHeight: uint32(height-j),`  then on line 617 `height--`  wont be required.  similarly for `historicalSpendDetails`